### ISSUE TITLE
Type for objects and arrays

### ISSUE THREAD

**Issue by BrandonWeber** [i18next/react-i18next#1057]
**Describe the bug**
I use the library for Next, but [apparently](https://github.com/isaachinman/next-i18next/issues/641) types are imported from `i18next` and `react-i18next`.

When using [objects and arrays](https://www.i18next.com/translation-function/objects-and-arrays) with standard types, I encountered an error:

> Property 'map' does not exist on type 'string'.

I tried using different types of library.

**To Reproduce**
[https://codesandbox.io/s/next-i18next-example-uv98b](https://codesandbox.io/s/next-i18next-example-uv98b)

**Comment by rosskevin** [comment]
The return value is not something that can be statically determined so you must provide the type as a parameter.

See https://github.com/i18next/i18next/blob/master/test/typescript/t.test.ts#L19

**Comment by BrandonWeber** [comment]
I think you need to provide more examples of how library works with Typescript in documentation. Like here:
[https://react.i18next.com/legacy-v9/withnamespaces#using-with-typescript](https://react.i18next.com/legacy-v9/withnamespaces#using-with-typescript)

Now I'm trying to understand an example from test, but it does not seem to be related to `.map` array.

**Comment by rosskevin** [comment]
If you cannot read a typescript definition, then you have identified the source of your problem.  Typescript definitions are self documenting, and a handwritten documentation page would only be a lossy representation.

TL;DR if you read the typescript definitions then you would not have filed two issues on github regarding this matter.  It is not a bug, it is a lack of understanding.

I suggest you spend more time looking directly at the typescript definitions and filling in any gaps in understanding that way as the most effective way for you to move forward.

As a matter of etiquette, I suggest that telling open source contributors what we _should_ be doing for you is an approach you might reconsider.

**Comment by matsgm** [comment]
If you've wandered in here looking for a solution...

```
            <ul>
              {(t('someList', { returnObjects: true, defaultValue: ['1', '2'] }) as []).map((item: any) => (
                <li>
                  <Typography>
                    {item}
                  </Typography>
                </li>
              ))}
            </ul>
```

**Comment by rosskevin** [comment]
@matsgm that is not the best way for your case and it is not demonstrated that way in the tests I mentioned above.  

Use `t<string[]>(...).map()` for your case.

**Comment by matsgm** [comment]
> @matsgm that is not the best way for your case and it is not demonstrated that way in the tests I mentioned above.
> 
> Use `t<string[]>(...).map()` for your case.

Ah, thanks :)

**Comment by Jiia** [comment]
@rosskevin The links you provided return 404. The answers suggested do not work anymore with the latest version of i18next. It causes error:
> Expected 3 type arguments, but got 1.

**Comment by Jiia** [comment]
It would be highly helpful if the solution was included in the documentation at https://www.i18next.com/translation-function/objects-and-arrays

**Comment by rosskevin** [comment]
> @rosskevin The links you provided return 404. The answers suggested do not work anymore with the latest version of i18next. It causes error:
> 
> > Expected 3 type arguments, but got 1.

@Jiia this issue is OVER 4 years old and I was linking to a test source file that was valid at the time.  Things change, look for an updated version of the test source file.  

If you think documentation should be updated, I suggest figure out what needs to be updated and submit a PR with updated documentation.

**Comment by Jiia** [comment]
I only pointed out that the answer is outdated and not working anymore, there's no need to get emotional about it.

Hundreds or thousands of people run into the same problems, and end up in these same old threads through Google. It's only helpful to re-open the discussion relating to the same issue if the source has been updated and the answers are not correct anymore.

For the record, the solution was helpfully provided by other contributors at https://github.com/i18next/i18next/issues/2151
